### ISSUE TITLE
[TF2] DemoUI editor broken -- VDMs aren't created/updated.

### ISSUE THREAD

**Issue by ssumichrast** [ValveSoftware/Source-1-Games#1073]
I'm attempting to edit a demo file recorded from SourceTV tonight.  I've loaded the file up, it's obviously huge so I'm using an action at tick 1 to skip to a tick much later in the file.  Once I've configured this and choose save, the VDM never gets created.

I then created a VDM using a text editor and specified the appropriate action.  I re-loaded the demo in my TF2 client and sure enough it clicked to the appropriate tick.  I opened the Editor window and it shows my action.  I edit the same action, press okay and Save.  Then I opened the VDM in the text editor and it was not updated.  I added a new action (place text on the screen) and saved -- VDM still not updated.  Just to be safe I stopped playback and closed TF2.  Re-opened and re-loaded the demo -- only thing that happened was my original edit (when I manually created the VDM) loaded and forwarded me to the appropriate tick.

I'm not sure if this broken during the steampipe conversion (maybe confused with the new directory structure?)  I don't think the functionality is broken, clearly it can load VDM just fine.  It seems like it's just failing to write the file out for some reason -- even if the file already exists.

**Comment by kisak-valve** [comment]
Hello @ssumichrast and @ElectricTriangle, is this issue still occurring?

**Comment by ElectricTriangle** [comment]
Yeah, it definitely still seems to be broken. No VDMs are created upon saving the event list.

**Comment by ElectricTriangle** [comment]
I just did a little amateur sleuthing and I figured out what the problem is. While the game does look in the proper location for the VDM (it would appear to be included in the search path),
`6554  stat64("/opt/steamdir/steamapps/common/Team Fortress 2/tf/demos/2016-12-23_18-08-26.vdm",  <unfinished ...>`

it also has an improper location in the search path.
`6554  stat64("/opt/steamdir/steamapps/common/Team Fortress 2/tf/tf/demos/2016-12-23_18-08-26.vdm",  <unfinished ...>`

That improper location is also where the game tries to write the file which then fails if /tf/tf/demos doesn't exist.
`6554  open("/opt/steamdir/steamapps/common/Team Fortress 2/tf/tf/demos/2016-12-23_18-08-26.vdm", O_WRONLY|O_CREAT|O_TRUNC, 0666 <unfinished ...>`

With the folder created, the VDM files are created properly and could probably be loaded from the same dir too. I imagine the only thing that needs to be done in the game code is to correct the erroneous "/tf/tf" directory.

**Comment by CobaltSpace** [comment]
An alternative solution, and the one I use is to create a symlink named `tf` targeting `.` in the `tf` folder.  I'm not sure if this a solution that can work for windows (or if it even is a problem on windows), but this is a fix on Linux.

```
cd "path/to/steamapps/common/Team Fortress 2/tf"
ln -s . tf
```
